ui/selectors/confirm-transaction.js

```diff
@@ -218,7 +206,7 @@ export const sendTokenTokenAmountAndToAddressSelector = createSelector(
 );
 
 export const contractExchangeRateSelector = createSelector(
-  contractExchangeRatesSelector,
+  (state) => getTokenExchangeRates(state),
```

Comment:
You'd think `getTokenExchangeRates` would work here instead of a lambda, but the tests don't like it.  I suspect that's why the original code duplicated the selector.

```
 FAIL  app/scripts/metamask-controller.actions.test.js
  ● Test suite failed to run

    Selector creators expect all input-selectors to be functions, instead received the following types: [undefined, function]

      206 | );
      207 |
    > 208 | export const contractExchangeRateSelector = createSelector(
          |                                                           ^
      209 |   getTokenExchangeRates,
```


I've encountered this situation a few times, and never discovered why it happens.  I think you can also fix it by stubbing the selector in the tests.  But the lambda seems to work too.

Reply:
This error is caused by the function being `undefined` at the point in time this is executed. This could happen because the function is defined later in the module (as a variable, so not hoisted to the top like function declarations are), but more often I've seen this happen due to a circular dependency between the current file and the file we're importing the selector from.

One of the tech debt tasks in-progress now is an effort to remove all circular dependencies in our codebase and implement a lint rule preventing their re-introduction, so hopefully we can resolve this soon. But in the meantime this solution works.